Buildings/Fluid/CHPs/BaseClasses/AssertPower.mo

```diff
@@ -0,0 +1,131 @@
+within Buildings.Fluid.CHPs.BaseClasses;
+model AssertPower "Assert if electric power is outside boundaries"
+  extends Modelica.Blocks.Icons.Block;
+
+  parameter Modelica.SIunits.Power PEleMax
+    "Maximum power output";
+  parameter Modelica.SIunits.Power PEleMin
+    "Minimum power output";
+  parameter Boolean dPEleLim
+    "If true, the rate at which net power output can change is limited";
+  parameter Real dPEleMax
+    "Maximum rate at which net power output can change in W/s";
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput PEleDem(
+    final unit="W")
+    "Electric power demand"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),
+      iconTransformation(extent={{-140,-20},{-100,20}})));
+  Buildings.Controls.OBC.CDL.Utilities.Assert assMesP(
+    final message="Electric power is outside boundaries!")
+    "Generate warning when the electric power demand is out of the range"
+    annotation (Placement(transformation(extent={{70,10},{90,30}})));
+  Buildings.Controls.OBC.CDL.Utilities.Assert assMesDP(
+    final message="Rate of change in power output is outside boundaries!")
+    "Assert function for checking power rate"
+    annotation (Placement(transformation(extent={{70,-50},{90,-30}})));
+
+protected
+  Buildings.Controls.OBC.CDL.Logical.Nor nor
+    "Check if the electric power demand is out of the power production range"
+    annotation (Placement(transformation(extent={{40,10},{60,30}})));
+  Buildings.Controls.OBC.CDL.Logical.Nand nand "Logical Nand"
+    annotation (Placement(transformation(extent={{40,-50},{60,-30}})));
+  Buildings.Controls.OBC.CDL.Logical.Sources.Constant cheDPLim(
+    final k=dPEleLim)
+    "Check if dP is limited"
+    annotation (Placement(transformation(extent={{0,-90},{20,-70}})));
```

Comment:
Removed all blocks that are not needed if `dpEleLim=false`. Otherwise you compute derivatives and inequalities simply to ignore the result.